cpp/oneapi/dal/algo/shortest_paths/test/badargs.cpp

```diff
@@ -0,0 +1,164 @@
+    }
+
+    std::int64_t get_edge_count() const {
+        return edge_count;
+    }
+
+    std::int64_t get_cols_count() const {
+        return cols_count;
+    }
+
+    std::int64_t get_rows_count() const {
+        return rows_count;
+    }
+
+    int get_source() const {
+        return source;
+    }
+
+protected:
+    std::int64_t vertex_count;
+    std::int64_t edge_count;
+    std::int64_t cols_count;
+    std::int64_t rows_count;
+    int source;
+};
+
+class example_graph_type : public graph_base_data {
+public:
+    example_graph_type() {
+        vertex_count = 6;
+        edge_count = 9;
+        cols_count = 9;
+        rows_count = 7;
+    }
+    std::array<std::int64_t, 7> rows = { 0, 2, 4, 6, 8, 9, 9 };
+    std::array<std::int32_t, 9> cols = { 1, 2, 3, 4, 3, 4, 4, 5, 5 };
+    std::array<double, 9> edge_weights = { 10, 20, 50, 10, 20, 33, 20, 2, 1 };
+};
+
+class empty_graph_type : public graph_base_data {
+public:
+    empty_graph_type() {
+        vertex_count = 0;
+        edge_count = 0;
+        cols_count = 0;
+        rows_count = 1;
+    }
+    std::array<std::int64_t, 1> rows = { 0 };
+    std::array<std::int32_t, 0> cols = {};
+    std::array<double, 0> edge_weights = {};
+};
+
+class negative_weights_graph_type : public graph_base_data {
+public:
+    negative_weights_graph_type() {
+        vertex_count = 3;
+        edge_count = 2;
+        cols_count = 2;
+        rows_count = 4;
+    }
+    std::array<std::int64_t, 4> rows = { 0, 1, 2, 2 };
+    std::array<std::int32_t, 2> cols = { 1, 2 };
+    std::array<double, 2> edge_weights = { -10, -20 };
+};
+
+class shortest_paths_badargs_test {
+public:
+    template <typename GraphType>
+    void check_shortest_paths(double delta, int source, bool nothing_to_calculate = false) {
+        using namespace dal::preview::shortest_paths;
+        GraphType graph_data;
+
+        const auto graph_builder = dal::preview::detail::directed_adjacency_vector_graph_builder<
+            int,
+            double,
+            oneapi::dal::preview::empty_value,
+            int,
+            std::allocator<char>>(graph_data.get_vertex_count(),
+                                  graph_data.get_edge_count(),
+                                  graph_data.rows.data(),
+                                  graph_data.cols.data(),
+                                  graph_data.edge_weights.data());
+
+        const auto& graph = graph_builder.get_graph();
+
+        auto result_type = nothing_to_calculate
+                               ? optional_results::distances & optional_results::predecessors
+                               : optional_results::distances | optional_results::predecessors;
+        std::allocator<char> alloc;
+
+        const auto shortest_paths_desc =
+            descriptor<float, method::delta_stepping, task::one_to_all, std::allocator<char>>(
+                source,
+                delta,
+                result_type,
+                alloc);
```

Comment:
Why do we need this?  std alloc is the default type and default constructor as a default value.